### PR TITLE
⬆️ Upgrading `dynamic-sidecar` dependencies

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

- 🐛 fixed pytest fixture which caused socketio `disconnect` events to not be triggered in the aiohttp app.
- 🐛 fixed failing test to fail more reliably (it was hanging and timing out the CI)

###  Highlights on updated libraries (only updated libraries are included)

- #packages before ~ 89
- #packages after ~ 89

|#|name|before|after|upgrade|count|packages|
|-|----|------|-----|-------|-----|--------|
|   1 | aio-pika                  | 9.1.2           | 9.4.1      | *minor*    | 1 | dynamic-sidecar⬆️ |
|   2 | aioboto3                  | 12.0.0          | 12.3.0     | *minor*    | 1 | dynamic-sidecar🧪 |
|   3 | aiobotocore               | 2.7.0           | 2.11.2     | *minor*    | 1 | dynamic-sidecar🧪 |
|   4 | aiocache                  | 0.12.1          | 0.12.2     |            | 1 | dynamic-sidecar⬆️ |
|   5 | aiofiles                  | 23.1.0          | 23.2.1     | *minor*    | 1 | dynamic-sidecar⬆️ |
|   6 | aiohttp                   | 3.8.5           | 3.9.3      | *minor*    | 3 | dynamic-sidecar⬆️🧪🔧 |
|   7 | aiormq                    | 6.7.7           | 6.8.0      | *minor*    | 1 | dynamic-sidecar⬆️ |
|   8 | aiosignal                 | 1.3.1           | 🗑️ removed |  | 1 | dynamic-sidecar🔧 |
|   9 | alembic                   | 1.11.1          | 1.13.1     | *minor*    | 1 | dynamic-sidecar⬆️ |
|  10 | anyio                     | 3.7.1           | 4.3.0      | **MAJOR**  | 1 | dynamic-sidecar⬆️ |
|  11 | arrow                     | 1.2.3           | 1.3.0      | *minor*    | 1 | dynamic-sidecar⬆️ |
|  12 | astroid                   | 3.0.2           | 3.1.0      | *minor*    | 1 | dynamic-sidecar🔧 |
|  13 | async-timeout             | 4.0.2           | 4.0.3      |            | 3 | dynamic-sidecar⬆️🧪🔧 |
|  14 | asyncpg                   | 0.28.0          | 0.29.0     | *minor*    | 1 | dynamic-sidecar⬆️ |
|  15 | attrs                     | 23.1.0          | 23.2.0     | *minor*    | 3 | dynamic-sidecar⬆️🧪🔧 |
|  16 | bidict                    | 0.22.1          | 0.23.1     | *minor*    | 1 | dynamic-sidecar⬆️ |
|  17 | black                     | 23.12.0         | 24.3.0     | **MAJOR**  | 1 | dynamic-sidecar🔧 |
|  18 | boto3                     | 1.28.64         | 1.34.34    | *minor*    | 1 | dynamic-sidecar🧪 |
|  19 | botocore                  | 1.31.64         | 1.34.34    | *minor*    | 1 | dynamic-sidecar🧪 |
|  20 | build                     | 1.0.3           | 1.2.1      | *minor*    | 1 | dynamic-sidecar🔧 |
|  21 | certifi                   | 2023.7.22       | 2024.2.2   | **MAJOR**  | 2 | dynamic-sidecar⬆️🧪 |
|  22 | charset-normalizer        | 3.2.0           | 3.3.2      | *minor*    | 3 | dynamic-sidecar⬆️🧪🔧 |
|  23 | click                     | 8.1.6           | 8.1.7      |            | 2 | dynamic-sidecar⬆️🔧 |
|  24 | coverage                  | 7.3.2           | 7.4.4      | *minor*    | 1 | dynamic-sidecar🧪 |
|  25 | dill                      | 0.3.7           | 0.3.8      |            | 1 | dynamic-sidecar🔧 |
|  26 | dnspython                 | 2.4.1           | 2.6.1      | *minor*    | 1 | dynamic-sidecar⬆️ |
|  27 | docker                    | 6.1.3           | 7.0.0      | **MAJOR**  | 1 | dynamic-sidecar🧪 |
|  28 | email-validator           | 2.0.0.post2     | 2.1.1      | *minor*    | 1 | dynamic-sidecar⬆️ |
|  29 | exceptiongroup            | 1.1.2           | 1.2.0      | *minor*    | 2 | dynamic-sidecar⬆️🧪 |
|  30 | faker                     | 19.13.0         | 24.4.0     | **MAJOR**  | 1 | dynamic-sidecar🧪 |
|  31 | filelock                  | 3.13.1          | 3.13.3     |            | 1 | dynamic-sidecar🔧 |
|  32 | flaky                     | 3.7.0           | 3.8.1      | *minor*    | 1 | dynamic-sidecar🧪 |
|  33 | frozenlist                | 1.4.0           | 1.4.1      |            | 3 | dynamic-sidecar⬆️🧪🔧 |
|  34 | greenlet                  | 2.0.2           | 3.0.3      | **MAJOR**  | 2 | dynamic-sidecar⬆️🧪 |
|  35 | httpcore                  | 0.17.3          | 1.0.5      | **MAJOR**  | 1 | dynamic-sidecar⬆️ |
|  36 | httpx                     | 0.24.1          | 0.27.0     | *minor*    | 1 | dynamic-sidecar⬆️ |
|  37 | identify                  | 2.5.33          | 2.5.35     |            | 1 | dynamic-sidecar🔧 |
|  38 | idna                      | 3.4             | 3.6        | *minor*    | 3 | dynamic-sidecar⬆️🧪🔧 |
|  39 | jsonschema                | 4.18.4          | 4.21.1     | *minor*    | 1 | dynamic-sidecar⬆️ |
|  40 | mako                      | 1.2.4           | 1.3.2      | *minor*    | 1 | dynamic-sidecar⬆️ |
|  41 | markupsafe                | 2.1.3           | 2.1.5      |            | 1 | dynamic-sidecar⬆️ |
|  42 | multidict                 | 6.0.4           | 6.0.5      |            | 3 | dynamic-sidecar⬆️🧪🔧 |
|  43 | mypy                      | 1.6.1           | 1.9.0      | *minor*    | 1 | dynamic-sidecar🧪 |
|  44 | orjson                    | 3.9.7           | 3.10.0     | *minor*    | 1 | dynamic-sidecar⬆️ |
|  45 | packaging                 | 23.1            | 24.0       | **MAJOR**  | 3 | dynamic-sidecar⬆️🧪🔧 |
|  46 | pamqp                     | 3.2.1           | 3.3.0      | *minor*    | 1 | dynamic-sidecar⬆️ |
|  47 | pint                      | 0.22            | 0.23       | *minor*    | 1 | dynamic-sidecar⬆️ |
|  48 | pip-tools                 | 7.3.0           | 7.4.1      | *minor*    | 1 | dynamic-sidecar🔧 |
|  49 | platformdirs              | 4.1.0           | 4.2.0      | *minor*    | 1 | dynamic-sidecar🔧 |
|  50 | pluggy                    | 1.3.0           | 1.4.0      | *minor*    | 1 | dynamic-sidecar🧪 |
|  51 | pre-commit                | 3.6.0           | 3.7.0      | *minor*    | 1 | dynamic-sidecar🔧 |
|  52 | prometheus-client         | 0.19.0          | 0.20.0     | *minor*    | 1 | dynamic-sidecar⬆️ |
|  53 | psutil                    | 5.9.5           | 5.9.8      |            | 1 | dynamic-sidecar⬆️ |
|  54 | psycopg2-binary           | 2.9.6           | 2.9.9      |            | 1 | dynamic-sidecar⬆️ |
|  55 | pydantic                  | 1.10.12         | 1.10.15    |            | 1 | dynamic-sidecar⬆️ |
|  56 | pygments                  | 2.15.1          | 2.17.2     | *minor*    | 1 | dynamic-sidecar⬆️ |
|  57 | pyinstrument              | 4.5.1           | 4.6.2      | *minor*    | 1 | dynamic-sidecar⬆️ |
|  58 | pylint                    | 3.0.3           | 3.1.0      | *minor*    | 1 | dynamic-sidecar🔧 |
|  59 | pytest                    | 7.4.3           | 8.1.1      | **MAJOR**  | 1 | dynamic-sidecar🧪 |
|  60 | pytest-cov                | 4.1.0           | 5.0.0      | **MAJOR**  | 1 | dynamic-sidecar🧪 |
|  61 | pytest-mock               | 3.12.0          | 3.14.0     | *minor*    | 1 | dynamic-sidecar🧪 |
|  62 | python-dateutil           | 2.8.2           | 2.9.0.post0 | *minor*    | 2 | dynamic-sidecar⬆️🧪 |
|  63 | python-dotenv             | 1.0.0           | 1.0.1      |            | 1 | dynamic-sidecar🧪 |
|  64 | python-engineio           | 4.8.2           | 4.9.0      | *minor*    | 1 | dynamic-sidecar⬆️ |
|  65 | python-socketio           | 5.11.0          | 5.11.2     |            | 1 | dynamic-sidecar⬆️ |
|  66 | redis                     | 4.6.0           | 5.0.3      | **MAJOR**  | 1 | dynamic-sidecar⬆️ |
|  67 | rich                      | 13.4.2          | 13.7.1     | *minor*    | 1 | dynamic-sidecar⬆️ |
|  68 | rpds-py                   | 0.9.2           | 0.18.0     | *minor*    | 1 | dynamic-sidecar⬆️ |
|  69 | ruff                      | 0.1.8           | 0.3.5      | *minor*    | 1 | dynamic-sidecar🔧 |
|  70 | s3transfer                | 0.7.0           | 0.10.1     | *minor*    | 1 | dynamic-sidecar🧪 |
|  71 | sniffio                   | 1.3.0           | 1.3.1      |            | 2 | dynamic-sidecar⬆️🧪 |
|  72 | sqlalchemy                | 1.4.49          | 1.4.52     |            | 2 | dynamic-sidecar⬆️🧪 |
|  73 | tenacity                  | 8.2.2           | 8.2.3      |            | 1 | dynamic-sidecar⬆️ |
|  74 | tomlkit                   | 0.12.3          | 0.12.4     |            | 1 | dynamic-sidecar🔧 |
|  75 | toolz                     | 0.12.0          | 0.12.1     |            | 1 | dynamic-sidecar⬆️ |
|  76 | tqdm                      | 4.65.0          | 4.66.2     | *minor*    | 1 | dynamic-sidecar⬆️ |
|  77 | typer                     | 0.9.0           | 0.12.1     | *minor*    | 1 | dynamic-sidecar⬆️ |
|  78 | types-aiobotocore-s3      | 2.7.0           | 2.12.2     | *minor*    | 1 | dynamic-sidecar🧪 |
|  79 | types-aiofiles            | 23.2.0.0        | 23.2.0.20240403 |            | 1 | dynamic-sidecar🧪 |
|  80 | types-pyyaml              | 6.0.12.12       | 6.0.12.20240311 |            | 1 | dynamic-sidecar🧪 |
|  81 | typing-extensions         | 4.7.1           | 4.11.0     | *minor*    | 3 | dynamic-sidecar⬆️🧪🔧 |
|  82 | uvicorn                   | 0.23.1          | 0.29.0     | *minor*    | 1 | dynamic-sidecar⬆️ |
|  83 | virtualenv                | 20.25.0         | 20.25.1    |            | 1 | dynamic-sidecar🔧 |
|  84 | watchdog                  | 3.0.0           | 4.0.0      | **MAJOR**  | 1 | dynamic-sidecar⬆️ |
|  85 | websocket-client          | 1.6.4           | 🗑️ removed |  | 1 | dynamic-sidecar🧪 |
|  86 | wheel                     | 0.42.0          | 0.43.0     | *minor*    | 1 | dynamic-sidecar🔧 |
|  87 | wrapt                     | 1.15.0          | 1.16.0     | *minor*    | 1 | dynamic-sidecar🧪 |
|  88 | yarl                      | 1.9.2           | 1.9.4      |            | 3 | dynamic-sidecar⬆️🧪🔧 |

*Legend*: 
- ⬆️ base dependency (only services because packages are floating)
- 🧪 test dependency
- 🔧 tool dependency

## Related issue/s

- https://github.com/miguelgrinberg/python-socketio/issues/1319
## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: services/dynamic-sidecar/requirements/constraints.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# From 3.9.0 the follwong test fails: `tests/unit/test_modules_system_monitor__notifier.py::test_notifier_publish_message`
+# the room disconnect event is no longer propagated
+# Issue triggered by:
+# - this change: https://github.com/ttsia/aiohttp/commit/70b3536e3eef1627bbb425e71faea2f8ce1c6f9b
+# - and it's PR: https://github.com/aio-libs/aiohttp/pull/7680
+# - this is the breaking change (1 line) https://github.com/ttsia/aiohttp/commit/70b3536e3eef1627bbb425e71faea2f8ce1c6f9b#diff-5eda97b86ffb5b1e8b797dbacc03e6d75767caada5d09956c1035b5684711965R204
+# submitted issue https://github.com/miguelgrinberg/python-socketio/issues/1319
+aiohttp<3.9.0
```
Comment (pcrespov) [inline services/dynamic-sidecar/requirements/constraints.txt:8]: Should we add this in the webserver as well? what about repo-wide?

**Review by pcrespov — APPROVED** [review]
thx

**Review by sanderegg — APPROVED** [review]

File: services/dynamic-sidecar/requirements/constraints.txt (hunk 1 above)
Comment (GitHK) [reply services/dynamic-sidecar/requirements/constraints.txt:8]: this only goes on services which use this feature. I'm currently processing some feedback. Still need to decide if this is an issue with how we test thing or with `python-socketio`.

**Comment by GitHK** [comment]
@pcrespov @sanderegg @matusdrobuliak66 you can now upgrade aiohttp, I have fixed addressed the issue with python-socketio

File: services/dynamic-sidecar/requirements/constraints.txt (hunk 1 above)
Comment (GitHK) [reply services/dynamic-sidecar/requirements/constraints.txt]: Thanks to some feedback from the maintainer of `python-socketio`, I was able to figure out where the issue resided in our codebase and fixed it.

**Pull request merged**